subscription/src/main/java/org/killbill/billing/subscription/api/svcs/DefaultSubscriptionInternalApi.java

```diff
@@ -318,7 +319,15 @@ public SubscriptionBaseBundle createBundleForAccount(final UUID accountId, final
         final DateTime now = clock.getUTCNow();
         final DateTime originalCreatedDate = existingBundles.size() > 0 ? existingBundles.get(0).getCreatedDate() : now;
         final DefaultSubscriptionBaseBundle bundle = new DefaultSubscriptionBaseBundle(bundleKey, accountId, now, originalCreatedDate, now, now);
-        return dao.createSubscriptionBundle(bundle, context);
+        SubscriptionBaseBundle subscriptionBundle = null;
+        try {
+            subscriptionBundle = dao.createSubscriptionBundle(bundle, context);
+        } catch (final Exception e) {
+            if (e.getCause() instanceof SQLDataException) {
+                throw new SubscriptionBaseApiException(e, ErrorCode.EXTERNAL_KEY_LIMIT_EXCEEDED);
+            }
```

Comment:
Any other exception is swallowed - you should re-throw instead (same comment elsewhere).